compiler/modules/front/resolveTypes.bal

```diff
@@ -39,6 +39,13 @@ function resolveTypes(ModuleSymbols mod) returns ResolveTypeError? {
     }
 }
 
+function resolveType(ModuleSymbols mod, string name) returns ResolveTypeError? {
+    s:ModuleLevelDefn? defn = mod.defns[name];
```

Comment:
This isn't doing what resolveTypes is doing...

I suggest you have a `resolveDefn(ModuleLevelDefn function)` which `resolveTypes` calls.